src/VisualStudio/Core/Def/ExternalAccess/VSTypeScript/Api/VSTypeScriptContainedLanguageWrapper.cs

```diff
@@ -71,24 +44,25 @@ internal struct VSTypeScriptContainedLanguageWrapper
             uint itemid,
             Guid languageServiceGuid)
         {
-            var filePath = ContainedLanguage.GetFilePathFromHierarchyAndItemId(hierarchy, itemid);
-            var projectId = ProjectId.CreateNewId($"Project for {filePath}");
-            workspace.OnProjectAdded(ProjectInfo.Create(projectId, VersionStamp.Default, filePath, string.Empty, InternalLanguageNames.TypeScript));
+            var projectId = ProjectId.CreateNewId();
 
             _underlyingObject = new ContainedLanguage(
                 bufferCoordinator,
                 componentModel,
                 workspace,
                 projectId,
                 null,
-                filePath,
+                hierarchy,
+                itemid,
                 languageServiceGuid,
                 vbHelperFormattingRule: null);
+
+            var filePath = _underlyingObject.GetFilePathFromBufferOrHierarchy(hierarchy, itemid);
```

Comment:
>var filePath = _underlyingObject.GetFilePathFromBufferOrHierarchy(hierarchy, itemid); [](http://example.com/codeflow?start=12&length=85)

Shouldn't this be preferring it off the textdocument too?

Reply:
oh, I see. You are calculating it again instead of having the ContainedLanguage expose it.

Reply:
Feel free to won't fix, but I think it would be cleaner to expose a filePath property on the ContainedLanguage rather than the GetFilePathFromBuffers method.